modules/sql-engine/src/main/java/org/apache/ignite/internal/sql/engine/metadata/ColocationGroup.java

```diff
@@ -194,30 +186,74 @@ public ColocationGroup colocate(ColocationGroup other) throws ColocationMappingE
         return new ColocationGroup(sourceIds, nodeNames, assignments);
     }
 
+    private List<NodeWithTerm> intersect(List<NodeWithTerm> assign0, List<NodeWithTerm> assign1, @Nullable Set<String> filter, int partId)
```

Comment:
'filter' can be Predicate and non-nullable.
So, for 'nodeNames == null' you can pass predicate '() -> true'

Reply:
Done, thanks